tests/unit/test_space.py

```diff
@@ -11,18 +11,47 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from __future__ import annotations
+
 import copy
 import itertools
 from typing import Container, FrozenSet, Tuple
 
 import numpy.testing as npt
 import pytest
 import tensorflow as tf
+from typing_extensions import Final
 
 from tests.util.misc import TF_DEBUGGING_ERROR_TYPES, ShapeLike, various_shapes
 from trieste.space import Box, DiscreteSearchSpace, SearchSpace
 
 
+class Integers(SearchSpace):
```

Comment:
I actually don't understand why we need this extra space for our tests.

Reply:
`__pow__` is defined on `SearchSpace`, so this lets us know that `__pow__` will work correctly if someone subclasses `SearchSpace` instead of using dss or box